### PR TITLE
Adds webpack eslint plugin.

### PR DESCRIPTION
**Pull request by flynnt**
`eslint-loader` was recently deprecated. This removes that and adds in `eslint-webpack-plugin`.
https://www.npmjs.com/package/eslint-webpack-plugin

It has a handful of improvements over the loader. It also doesn't break reloads on errors, so we can set `no-console` back to an error, but will still be able to console out as much as we want during development without having to restart things.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ module.exports = {
   ],
   "parser": "babel-eslint",
   "rules": {
-    "no-console": "off",
+    "no-console": 2,
```
Comment (brianjhanson) [inline .eslintrc.js:8]: Curious about this change, the docs usually recommend the string versions because they're much less obscure

**Comment by flynnt** [comment]
Oh. No real preference. I thought integers were preferred, but I guess I'm
wrong.

File: .eslintrc.js (hunk 1 above)
Comment (chrismlusk) [reply .eslintrc.js:8]: +1 strings for clarity

**Review by brianjhanson — APPROVED** [review]

File: .eslintrc.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ module.exports = {
   ],
   "parser": "babel-eslint",
   "rules": {
-    "no-console": "off",
+    "no-console": "error",
```
Comment (brianjhanson) [inline .eslintrc.js:8]: @flynnt I just noticed this, why did this get turned back on? 

https://github.com/onedesign/craft-starter/pull/39#discussion_r468662209

Comment (flynnt) [reply .eslintrc.js:8]: Well, I thought the new plugin wouldn't break the watch task if console was set to `error`. Turns out I was wrong. We should change it back to `warn`.

**Pull request merged**